### PR TITLE
TrackbarWrapper

### PR DESCRIPTION
**Pull request by BlinkovaEkaterina**
@vasily-v-ryabov 
I'd like to ask you some questions.
I added basic methods (based on basic messages) to TrackbarWrapper and covered it with tests.
If it is not enough, please, give me list with all needed messages which I should include.

### CODE REVIEW

**Comment by vasily-v-ryabov** [comment]
@BlinkovaEkaterina, I see you've just restored commented methods. We also need methods handling the following messages (as a high priority):
```
TBM_GETCHANNELRECT
TBM_GETLINESIZE / TBM_SETLINESIZE
TBM_GETPAGESIZE / TBM_SETPAGESIZE
TBM_GETTOOLTIPS
TBM_SETSEL / TBM_GETSELSTART / TBM_GETSELEND
```
`TBS_ENABLESELRANGE` should be checked in method `is_selection_range` using `self.style()` return value.

Other messages are low priority and not necessary to implement.

File: pywinauto/controls/common_controls.py [hunk 1]
```diff
@@ -3129,41 +3129,47 @@ def decrement(self):
...
-#    # Non PEP-8 alias
-#    GetRangeMin = get_range_min
-#
-#    #----------------------------------------------------------------
-#    def get_tool_tips_control(self):
-#        "Return the tooltip control associated with this control"
-#        return ToolTipsWrapper(self.send_message(win32defines.TBM_GETTOOLTIPS))
+    def get_range_min(self):
+        """Get min available trackbar value"""
+        return self.SendMessage(win32defines.TBM_GETRANGEMIN)
```
Comment (vasily-v-ryabov) [inline pywinauto/controls/common_controls.py]: (minor) I'd recommend using PEP8 compatible name `send_message`.

File: pywinauto/controls/common_controls.py [hunk 2]
```diff
@@ -3129,41 +3129,47 @@ def decrement(self):
...
+
+    def set_range_min(self, range_min):
+        """Set min available trackbar value"""
+        if range_min > self.get_range_max():
+            raise RuntimeError('Failed to set range min')
+        self.SendMessage(win32defines.TBM_SETRANGEMIN, True, range_min)
+
+    def set_position(self, pos):
+        """Set trackbar position"""
+        if self.get_range_max() < pos or self.get_range_min() > pos:
```
Comment (vasily-v-ryabov) [inline pywinauto/controls/common_controls.py]: Could be simplified:
```python
if not (self.get_range_min() <= pos <= self.get_range_max()):
    raise ValueError('Cannot set position out of range')
```

File: pywinauto/controls/common_controls.py [hunk 3]
```diff
@@ -3129,42 +3129,79 @@ def decrement(self):
...
+    def set_range_min(self, range_min):
+        """Set min available trackbar value"""
+        if range_min > self.get_range_max():
+            raise RuntimeError('Failed to set range min')
+        self.send_message(win32defines.TBM_SETRANGEMIN, True, range_min)
+
+    def set_position(self, pos):
+        """Set trackbar position"""
+        if not (self.get_range_min() <= pos <= self.get_range_max()):
+          raise ValueError('Cannot set position out of range')
```
Comment (vasily-v-ryabov) [inline pywinauto/controls/common_controls.py]: Incorrect indentation.

File: pywinauto/controls/common_controls.py [hunk 4]
```diff
@@ -3129,42 +3129,79 @@ def decrement(self):
...
+
+        return system_rect
+
+    def get_line_size(self):
+        """Get the number of logical positions the trackbar's slider"""
+        return self.send_message(win32defines.TBM_GETLINESIZE)
+
+    def get_tooltips_control(self):
+        "Get trackbar tooltip"
+        return self.send_message(win32defines.TBM_GETTOOLTIPS)
```
Comment (vasily-v-ryabov) [inline pywinauto/controls/common_controls.py]: Could be wrapped by `TooltipsWrapper` (like it's done in `ToolbarWrapper`).

File: pywinauto/controls/common_controls.py [hunk 5]
```diff
@@ -3129,42 +3129,79 @@ def decrement(self):
...
+
+    def get_page_size(self):
+        """Get the number of logical positions the trackbar's slider """
+        return self.send_message(win32defines.TBM_GETPAGESIZE)
+
+    def set_range_max(self, range_max):
+        """Set max available trackbar value"""
+        if range_max < self.get_range_min():
+            raise RuntimeError('Failed to set range max')
```
Comment (vasily-v-ryabov) [inline pywinauto/controls/common_controls.py]: Should be `ValueError` as well.

Comment (vasily-v-ryabov) [inline pywinauto/controls/common_controls.py]: `raise ValueError('Cannot set range max less than range min')`

**Comment by BlinkovaEkaterina** [comment]
@vasily-v-ryabov, please, rewiev
Test application does not have TrackbarWrapper tooltip, if I get it clear. That`s why I tested it with assertRaises.

File: pywinauto/controls/common_controls.py [hunk 6]
```diff
@@ -3129,42 +3129,100 @@ def decrement(self):
...
+        del remote_mem
+
+        return system_rect
+
+    def get_line_size(self):
+        """Get the number of logical positions the trackbar's slider"""
+        return self.send_message(win32defines.TBM_GETLINESIZE)
+
+    def get_tooltips_control(self):
+        "Get trackbar tooltip"
```
Comment (vasily-v-ryabov) [inline pywinauto/controls/common_controls.py]: Use triple double quotes for the docstring.

File: pywinauto/controls/common_controls.py [hunk 7]
```diff
@@ -3129,42 +3129,100 @@ def decrement(self):
...
+        return ToolTipsWrapper(self.send_message(win32defines.TBM_GETTOOLTIPS))
+
+    def get_page_size(self):
+        """Get the number of logical positions the trackbar's slider """
```
Comment (vasily-v-ryabov) [inline pywinauto/controls/common_controls.py]: Typo: should be `logical positions for the trackbar's slider` here and beyond.

File: pywinauto/controls/common_controls.py [hunk 8]
```diff
@@ -3129,42 +3129,100 @@ def decrement(self):
...
+            raise RuntimeError('Selection does not support for this TrackbarWrapper')
+        sel_start_val = win32functions.LoWord(sel_start)
+        sel_end_val = win32functions.HiWord(sel_end)
+        sel_val = win32functions.MakeLong(sel_start_val, sel_end_val)
+        self.send_message(win32defines.TBM_SETSAL, 0, sel_val)
+
+    def get_sel_start(self):
+        """Get start of selection"""
+        if not self.has_style(win32defines.TBS_ENABLESELRANGE):
+            raise RuntimeError('Selection does not support for this TrackbarWrapper')
```
Comment (vasily-v-ryabov) [inline pywinauto/controls/common_controls.py]: Grammar: should be `Range selection is not supported for this trackbar`

File: pywinauto/unittests/test_common_controls.py [hunk 9]
```diff
@@ -1500,5 +1501,104 @@ def testDecrement(self):
...
+        """Test the get_num_ticks method"""
+        self.assertEqual(self.ctrl.get_num_ticks(), 6)
+
+    def test_get_channel_rect(self):
+        """Test the get_channel_rect method"""
+        system_rect = win32structures.RECT()
+        system_rect.left = 8
+        system_rect.top = 19
+        system_rect.right = 249
+        system_rect.bottom = 23
```
Comment (vasily-v-ryabov) [inline pywinauto/unittests/test_common_controls.py]: Not sure about these magic numbers for the rect. How did you get them?

**Comment by vasily-v-ryabov** [comment]
Still fails on my laptop:
```
FAIL: Test the get_channel_rect method
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".\pywinauto\unittests\test_common_controls.py", line 1567, in test_get_channel_rect
    self.assert_channel_rect(self.ctrl.get_channel_rect(), system_rect)
  File ".\pywinauto\unittests\test_common_controls.py", line 1571, in assert_channel_rect
    self.assertEqual(first_rect.height, second_rect.height)
AssertionError: <bound method RECT.height of <RECT L8, T18, R249, B22>> != <bound method RECT.height of <RECT L8, T19, R249, B23>>

----------------------------------------------------------------------
Ran 125 tests in 82.648s

FAILED (failures=1)
```

**Pull request merged**